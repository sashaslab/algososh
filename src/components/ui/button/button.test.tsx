import { render, screen, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer'
import { Button } from "./button";

describe('Button test', () => {
    test('button with text', () => {
        const tree = renderer
        .create(<Button text="text" />)
        .toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('button without text', () => {
        const tree = renderer
        .create(<Button />)
        .toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('button disabled', () => {
        const tree = renderer
        .create(<Button disabled={true} />)
        .toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('button loader', () => {
        const tree = renderer
        .create(<Button isLoader={true} />)
        .toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('correct callback when clicking on a button', () => {
        const onClick = jest.fn();
        render(<Button text="click me" onClick={onClick} />)
        const button = screen.getByText("click me");
        fireEvent.click(button)
        expect(onClick).toHaveBeenCalled()
    })
})
