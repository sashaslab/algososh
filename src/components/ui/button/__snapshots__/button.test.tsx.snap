// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test button loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button test button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
